<?xml version="1.0"?>
<launch>
    <!-- topic name -->
    <arg name="scan_topic_name" default="/scan" />
    <arg name="map_topic_name" default="/map" />
    <arg name="odom_topic_name" default="roomba/odometry" />
    <arg name="est_pose_topic_name" default="/est_pose" />
    <arg name="est_poses_topic_name" default="/est_poses" />
    <arg name="obj_topic_name" default="/object_positions" />

    <!-- frame_id -->
    <arg name="map_frame_id" default="map" />
    <arg name="odom_frame_id" default="odom" />
    <arg name="base_link_frame_id" default="base_link" />
    <arg name="lidar_frame_id" default="/scan" />
    <arg name="thetas_frame_id" default="theta_s" />
    <arg name="dynamixel_frame_id" default="dynamixel" />
    <arg name="realsense_frame_id" default="realsense" />

    <!-- map server -->
    <arg name="map_file" default="$(find mcl)/mapdata/map.yaml" />
    <node pkg="map_server" type="map_server" name="global_map" args="$(arg map_file)" />

    <!-- marker update checker -->
	<node pkg="map_updater" type="map_update_checker" name="map_update_checker" output="screen" >
        <param name="file_name" value="arrange_1.csv" />
        <param name="marker_frame_id" value="$(arg map_frame_id)"/>
        <param name="est_pose_topic_name" value="$(arg est_pose_topic_name)" />
        <param name="obj_topic_name" value="$(arg obj_topic_name)" />
        <rosparam command="load" file="$(find map_updater)/config/object/object_list.yaml"/>
    </node>

    <!-- mcl -->
    <node pkg="mcl" type="mcl_node" name="mcl_node" >
        <param name="scan_topic_name" value="$(arg scan_topic_name)" />
        <param name="map_topic_name" value="$(arg map_topic_name)" />
        <param name="est_pose_topic_name" value="$(arg est_pose_topic_name)" />
        <param name="est_poses_topic_name" value="$(arg est_poses_topic_name)" />
        <param name="map_frame_id" value="$(arg map_frame_id)" />
        <param name="odom_frame_id" value="$(arg odom_frame_id)" />
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
        <rosparam command="load" file="$(find mcl)/config/mcl/mcl.yaml" />
    </node>

    <!-- tf -->
    <node pkg="mcl" type="tf_broadcaster_node" name="tf_broadcaster_node" >
        <param name="odom_topic_name" value="$(arg odom_topic_name)" />
        <param name="odom_frame_id" value="$(arg odom_frame_id)" />
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
        <param name="lidar_frame_id" value="$(arg lidar_frame_id)" />
        <param name="thetas_frame_id" value="$(arg thetas_frame_id)" />
        <param name="dynamixel_frame_id" value="$(arg dynamixel_frame_id)" />
        <param name="realsense_frame_id" value="$(arg realsense_frame_id)" />
        <rosparam command="load" file="$(find mcl)/config/tf/tf.yaml" />
    </node>

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find map_updater)/config/rviz/map_update_checker.rviz" />

</launch>